### ISSUE TITLE
Identifier 'e' has already been declared

### ISSUE THREAD

**Issue by highlabs** [Ziggeo/ziggeo-client-sdk#18]
Hi! o/

I'm trying to use `react-ziggeo` in my Next.js project. But I getting a error in `ziggeo-client-sdk`:

`./node_modules/ziggeo-client-sdk/build/ziggeo.js 11:24462
Module parse failed: Identifier 'e' has already been declared (11:24462)
You may need an appropriate loader to handle this file type.`

(full error in: [https://pastebin.com/aSq8ZSdk](https://pastebin.com/aSq8ZSdk))

Any ideas whats happening?

**Comment by sambua** [comment]
Hello @highlabs ,

Could you please advice which version of `react-ziggeo` and `ziggeo-client-sdk` you're using? And if possible provide with options you're using so we could reproduce this bug.

Thanks in advance

**Comment by highlabs** [comment]
Versions:

- `react-ziggeo`: 1.6.4
- `ziggeo-client-sk`: 2.33.0

And I'm using Webpack 4.19.1.

**Comment by highlabs** [comment]
@sambua any news?

**Comment by sambua** [comment]
Hello @highlabs,

Apologize, there was some misunderstanding, I thought I've replied to your message.

I just want to clarify because we don't have `react-ziggeo 1.6.4` version. Could you please recheck it?

**Comment by highlabs** [comment]
> Hello @highlabs,
> 
> Apologize, there was some misunderstanding, I thought I've replied to your message.
> 
> I just want to clarify because we don't have `react-ziggeo 1.6.4` version. Could you please recheck it?

Oh, sorry! My bad.

I tried again today, it's my package json:
```
{
  "name": "next-app",
  "description": "Next App",
  "version": "0.0.1",
  "scripts": {
    "lint": "xo",
    "dev": "NODE_ENV=development nodemon -w server.js server.js",
    "build": "next build",
    "clean": "rm -rf ./static/workbox ./static/manifest .next",
    "start": "NODE_ENV=production node server.js",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook_build": "build-storybook -c .storybook -o .out"
  },
  "dependencies": {
    "axios": "^0.18.0",
    "babel-eslint": "^8.2.6",
    "babel-plugin-inline-react-svg": "0.4.0",
    "boxicons": "^1.7.1",
    "cookie": "^0.3.1",
    "date-fns": "^1.29.0",
    "dotenv": "^6.0.0",
    "eslint": "^5.3.0",
    "eslint-config-airbnb": "^17.0.0",
    "eslint-plugin-import": "^2.13.0",
    "eslint-plugin-jsx-a11y": "^6.1.1",
    "eslint-plugin-react": "^7.10.0",
    "isomorphic-fetch": "^2.2.1",
    "jwt-decode": "^2.2.0",
    "next": "7.0.0",
    "next-images": "^0.10.6",
    "next-manifest": "highlabs/next-manifest",
    "next-page-transitions": "^1.0.0-alpha.4",
    "prop-types": "^15.6.2",
    "react": "^16.4.2",
    "react-dom": "^16.4.2",
    "react-facebook-login": "^4.1.0",
    "react-google-login": "^3.2.1",
    "react-modal": "^3.5.1",
    "react-player": "^1.6.4",
    "react-ziggeo": "^3.1.0",
    "unistore": "^3.0.6"
  },
  "devDependencies": {
    "@babel/core": "^7.1.0",
    "@babel/plugin-proposal-decorators": "^7.1.0",
    "@storybook/addon-options": "^3.4.10",
    "@storybook/react": "^3.4.10",
    "babel-plugin-react-require": "^3.0.0",
    "dotenv-webpack": "^1.5.7",
    "next-workbox": "^1.4.0",
    "node-sass": "^4.9.2",
    "nodemon": "^1.18.3",
    "styled-jsx-plugin-sass": "^0.2.4",
    "webpack": "^4.19.1",
    "workbox-webpack-plugin": "^3.4.1"
  }
}

```

`react-ziggeo` version: ^3.1.0

Error: https://pastebin.com/QTFwRdG7

**Comment by highlabs** [comment]
@sambua I created a new repo with empty project for testing: https://github.com/highlabs/next-ziggeo

When you run `$ yarn dev` you get the same error.

**Comment by sambua** [comment]
@highlabs ,
 
We could determine issue source and will update in the next update. 

Only one, point solving this exact issue, seems not working correctly in your case. As I understood your app is server-side renderer, so global variables like `document`, `navigator` etc. are not accessible. Which are required in Ziggeo app and core part.

Only way makes Ziggeo work is, render it in the client-side.

**Comment by highlabs** [comment]
@sambua ,

I tried to import in client-side only, but still getting the error.

The next.js framework has this feature: https://github.com/zeit/next.js/#3-with-no-ssr
(Updated the repo, using dynamic import: https://github.com/highlabs/next-ziggeo/commit/8d8a4f5e81256c93b253297d216ffae54bf99808 )

Maybe the problem are in the Webpack 4. Because I tested `react-ziggeo` with Next.js 6 (Webpack 3) and `next/dynamic` and it worked without problems. :)

I created a new repo, using `create-react-app` 2. (the new version use Webpack 4)
https://github.com/highlabs/ziggeo-cra
I get the same error: https://pastebin.com/t8dAtn6b

**Comment by sambua** [comment]
Hello @highlabs ,

Yes, for know it could show error, today later will push new react-ziggeo version, where will fix this bug. After will test with repo you provided, and let you know.

Regarding Webpack version, 4th version probably more smarter to detect this type of bugs :)

Thanks

**Comment by sambua** [comment]
@highlabs ,

Could you please check newly published version `react-ziggeo 3.1.1`? 

I've tested with `create-react-app` with SSL run with `HTTPS=true npm start` and seems now it works OK.

**Comment by highlabs** [comment]
It's worked with Next.js too! :) Thanks @sambua !

https://next-simple-starter-srbathiszc.now.sh/

(my app are broken, but it's not ziggeo problem. xD)
